client/pages/add-exercises.jsx

```diff
@@ -5,33 +5,108 @@ export default class AddExercisePage extends React.Component {
     super(props);
     this.state = {
       exerciseName: '',
-      exercises: []
+      exercises: [],
+      data: {
+        exercises: []
+      }
     };
     this.handleExerciseNameChange = this.handleExerciseNameChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
     this.handleClick = this.handleClick.bind(this);
+    this.handleRepsChange = this.handleRepsChange.bind(this);
+    this.handleWeightChange = this.handleWeightChange.bind(this);
+    this.handleIconChange = this.handleIconChange.bind(this);
   }
 
   handleExerciseNameChange(event) {
     this.setState({ exerciseName: event.target.value });
   }
 
+  handleRepsChange(index, event) {
+    const dataCopy = Object.assign({}, this.state.data);
+    dataCopy.exercises[index].reps = event.target.value;
+    this.setState({ data: dataCopy });
+  }
+
+  handleWeightChange(index, event) {
+    const dataCopy = Object.assign({}, this.state.data);
+    dataCopy.exercises[index].weight = event.target.value;
+    this.setState({ data: dataCopy });
+  }
+
+  handleIconChange(index, event) {
+    const dataCopy = Object.assign({}, this.state.data);
+    dataCopy.exercises[index].isCompleted = !dataCopy.exercises[index].isCompleted;
+    this.setState({ data: dataCopy });
+    if (!this.state.data.exercises[index].isCompleted) {
+      event.target.className = 'fa-regular fa-circle-check fa-lg';
+    } else {
+      event.target.className = 'fa-solid fa-circle-check fa-lg';
+    }
```

Comment:
noooo, we do not do dom manipulation like this in React apps. 😢 instead, pass the correct className prop to the react element in render()